### DISCUSSION TITLE
Welcome to Hapi Discussions!

### DISCUSSION THREAD

**Discussion by MAfarrag** [serapeum-org/Hapi#48] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by woaibianchen** [comment]
First of all, thank you for sharing. There is a question in running calibration.py file,  which is the system prompts that the ’'oasis'’ package is missing. Then, I found that the ‘’oasis‘’ package I downloaded does not have ''oasis.optimization''. I wonder if the author used a specific oasis package？My python version is 3.9 and oasis version is 0.1.3. I look forward to your answer.

**Reply by MAfarrag** [reply]
Hi @woaibianchen please install the latest version of oasis which is 1.0.2 and the latest version of Hapi (1.2.0) and try the examples here [Hydrological Model Examples](https://github.com/MAfarrag/Hapi/tree/hydraulic-model/examples/Hydrological%20model) everything should be working with these latest versions

**Reply by woaibianchen** [reply]
Thank you very much for your advice! I have successfully run the example, but there's another problem that bothers me. I tried to substitute the flow accumulation raster file of a 30000 km2 watershed into the model, but found that the model reported the error "flow accumulation raster values are not correct". I guess it's because not all grids flow out, but I've used ArcGIS to fill the DEM and generate ACC files. I don't know if there is any other way to deal with DEM and ensure that the maximum value of ACC file meets the standard?

**Reply by MAfarrag** [reply]
The Flow accumulation raster is the most important input as it tells the model whether the or not all the water goes to the outlet , so if number of cells inside the domain for example 100, the value of outlet cell in the flow accumulation raster has to be 100 and that is the check that Hapi does before starting the run

So to make sure that the Flow accumulation raster is correct you can follow this video ([Burning the River into DEM](https://www.youtube.com/watch?v=jHT_StPb_oM&ab_channel=HansvanderKwast)) however it uses QGIS you can do the same procedures in arc map
